src/components/Model3DViewer/Model3DViewer.tsx

```diff
@@ -235,21 +294,157 @@ export class Model3DViewer extends React.Component<Model3DViewerProps> {
     }
   };
 
+  slice = (sliceProps: SlicingProps) => {
+    if (this.viewer) {
+      const planes = this.state.planes;
+      if (sliceProps.x) {
+        const plane = new THREE.Plane(
+          new THREE.Vector3(sliceProps.x.direction ? 1 : -1, 0, 0),
+          sliceProps.x.coord
+        );
+        planes[0] = plane;
+      }
+      if (sliceProps.y) {
+        const plane = new THREE.Plane(
+          new THREE.Vector3(0, sliceProps.y.direction ? 1 : -1, 0),
+          sliceProps.y.coord
+        );
+        planes[1] = plane;
+      }
+      if (sliceProps.z) {
+        const plane = new THREE.Plane(
+          new THREE.Vector3(0, 0, sliceProps.z.direction ? 1 : -1),
+          sliceProps.z.coord
+        );
+        planes[2] = plane;
+      }
+      this.setState({ planes });
+    }
+  };
+
+  takeScreenshot = async () => {
+    if (this.viewer) {
+      const url = await this.viewer.getScreenshot();
+      if (this.props.onScreenshot) {
+        this.props.onScreenshot(url);
+      }
+    }
+  };
+
+  onChange = (val: number, axis: number) => {
+    const planes = this.state.planes;
+    const flipped = this.state.flipped;
+    planes[axis].set(planes[axis].normal, flipped[axis] ? val : -val);
+    this.setState({ planes });
+  };
+
+  flipSlider = (axis: number) => {
+    const planes = this.state.planes;
+    const flipped = this.state.flipped;
+    planes[axis].negate();
+    flipped[axis] = !flipped[axis];
+    this.setState({ planes });
+    this.setState({ flipped });
+  };
+
+  renderSlider = (range: SliderRange | undefined, axis: number) => {
+    if (!range) {
+      return <></>;
+    }
+    return (
+      <div style={containerStyles}>
+        <span style={{ marginTop: '0.5vh' }}>
+          <h4>x</h4>
+        </span>
+        <Slider
+          step={(range.max - range.min) / 100}
+          min={range.min}
+          max={range.max}
+          defaultValue={range.max}
+          style={{ width: '18vw' }}
+          onChange={(val: SliderValue) => this.onChange(val as number, axis)}
+        />
+        <Button
+          type="primary"
+          icon="redo"
+          size="small"
+          style={{ marginLeft: '2%' }}
+          onClick={() => this.flipSlider(axis)}
+        />
+      </div>
+    );
+  };
+
+  renderSliders = () => {
+    if (!this.props.slider) {
+      return null;
+    }
+    const xSlider = this.renderSlider(this.props.slider.x, 0);
+    const ySlider = this.renderSlider(this.props.slider.x, 0);
```

Comment:
Great, but all of these 3 lines uses `this.props.slider.x`...

Reply:
Fixed